src/Orleans.CodeGeneration/GrainReferenceGenerator.cs

```diff
@@ -209,15 +209,21 @@ private static MemberDeclarationSyntax[] GenerateInvokeMethods(Type grainType, A
 
                     if (isOneWayTask)
                     {
-                        if (method.ReturnType != typeof(Task))
+                        if (method.ReturnType == typeof(Task))
+                        {
+                            var done = typeof(Task).GetNameSyntax(true).Member((object _) => Task.CompletedTask);
+                            body.Add(SF.ReturnStatement(done));
+                        }
+                        else if (method.ReturnType == typeof(ValueTask))
+                        {
+                            body.Add(SF.ReturnStatement(SF.ParseExpression("default(System.Threading.Tasks.ValueTask)")));   
```

Comment:
I know this is not pretty. 
Maybe you guys can guide me on how to fix this?

Reply:
Would this work?

```c#
body.Add(SF.ReturnStatement(SF.DefaultExpression(typeof(ValueTask).GetTypeSyntax())))
```

Reply:
Tests are passing, yes.
Also, checked generated code. It's correct.

Reply:
example:
```csharp

global::System.Threading.Tasks.Task global::UnitTests.GrainInterfaces.IOneWayGrain.Notify(global::UnitTests.GrainInterfaces.ISimpleGrainObserver observer0)
{
	global::Orleans.CodeGeneration.GrainFactoryBase.CheckGrainObserverParamInternal(observer0);
	base.InvokeOneWayMethod(unchecked((int)0x92B0AD04), new object[]{observer0 is global::Orleans.Grain ? observer0.AsReference<global::UnitTests.GrainInterfaces.ISimpleGrainObserver>() : observer0});
	return global::System.Threading.Tasks.Task.CompletedTask;
}

global::System.Threading.Tasks.ValueTask global::UnitTests.GrainInterfaces.IOneWayGrain.NotifyValueTask(global::UnitTests.GrainInterfaces.ISimpleGrainObserver observer0)
{
	global::Orleans.CodeGeneration.GrainFactoryBase.CheckGrainObserverParamInternal(observer0);
	base.InvokeOneWayMethod((int)0x7F9DCB82, new object[]{observer0 is global::Orleans.Grain ? observer0.AsReference<global::UnitTests.GrainInterfaces.ISimpleGrainObserver>() : observer0});
	return default(System.Threading.Tasks.ValueTask);
}


```

Reply:
Can you push the changed to fix this here and below?

Reply:
Done. 
Generated code is like this now:

```csharp

global::System.Threading.Tasks.Task global::UnitTests.GrainInterfaces.IOneWayGrain.Notify(global::UnitTests.GrainInterfaces.ISimpleGrainObserver observer0)
{
	global::Orleans.CodeGeneration.GrainFactoryBase.CheckGrainObserverParamInternal(observer0);
	base.InvokeOneWayMethod(unchecked((int)0x92B0AD04), new object[]{observer0 is global::Orleans.Grain ? observer0.AsReference<global::UnitTests.GrainInterfaces.ISimpleGrainObserver>() : observer0});
	return global::System.Threading.Tasks.Task.CompletedTask;
}

global::System.Threading.Tasks.ValueTask global::UnitTests.GrainInterfaces.IOneWayGrain.NotifyValueTask(global::UnitTests.GrainInterfaces.ISimpleGrainObserver observer0)
{
	global::Orleans.CodeGeneration.GrainFactoryBase.CheckGrainObserverParamInternal(observer0);
	base.InvokeOneWayMethod((int)0x7F9DCB82, new object[]{observer0 is global::Orleans.Grain ? observer0.AsReference<global::UnitTests.GrainInterfaces.ISimpleGrainObserver>() : observer0});
	return default;
}


```